phylogeny/prototypeSelection.py

```diff
@@ -0,0 +1,99 @@
+"""
+Prototype selection
+
+Given a set of n elements (S) and their pairwise distances in terms of a
+skbio distance matrix (DM) and a given number (k << n), find the sub-set (s)
+consisting of exactly k elements, such that s best represents the full set S.
+
+Here, we define "best represents" as maximizing the sum of distances between
+all points, i.e. sum {over a,b in S} DM[a,b]. This is our objective function.
+
+This problem is known to be NP-hard [1], thus we need to resort to heuristics.
+In the future, this module will implement several heuristics, whose quality can
+be measured by the objective function for each problem instance, since there is
+no global winner.
+For completeness, the exact but exponential algorithm is implemented, too.
+  "prototypeSelection_exhaustive"
+
+[1] Gamez, J. Esteban, François Modave, and Olga Kosheleva.
+    "Selecting the most representative sample is NP-hard:
+     Need for expert (fuzzy) knowledge."
+    Fuzzy Systems, 2008. FUZZ-IEEE 2008.
+"""
+
+from skbio.stats.distance import DistanceMatrix
+import itertools
+import numpy as np
+import scipy as sp
+from typing import Sequence, Tuple
+
+
+def distanceSum(elements: Sequence[str], dm: DistanceMatrix) -> float:
```

Comment:
Minor but [PEP8](https://www.python.org/dev/peps/pep-0008/#function-names) defines variable and method names to use lowercase words separated by underscores, rather than mixedCase.
`distanceSum -> distance_sum`. Applies to all instances in the code (i.e. I will not comment this issue downstream)